mpas_analysis/shared/timekeeping/utility.py

```diff
@@ -0,0 +1,229 @@
+"""
+Time keeping utility functions
+
+Author
+------
+Xylar Asay-Davis
+
+Last Modified
+-------------
+02/06/2017
+"""
+
+import datetime
+from .MpasRelativeDelta import MpasRelativeDelta
+
+
+def stringToDatetime(dateString):  # {{{
+    """
+    Given a date string and a calendar, returns a `datetime.datetime`
+
+    Parameters
+    ----------
+    dateString : string
+        A date and time in one of the following formats:
+        - YYYY-MM-DD hh:mm:ss
+        - YYYY-MM-DD hh.mm.ss
+        - YYYY-MM-DD SSSSS
+        - DDD hh:mm:ss
+        - DDD hh.mm.ss
+        - DDD SSSSS
+        - hh.mm.ss
+        - hh:mm:ss
+        - YYYY-MM-DD
+        - YYYY-MM
+        - SSSSS
+
+        Note: either underscores or spaces can be used to separate the date
+        from the time portion of the string.
+
+    Returns
+    -------
+    datetime : A `datetime.datetime` object
+
+    Raises
+    ------
+    ValueError
+        If an invalid `dateString` is supplied.
+
+    Author
+    ------
+    Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/04/2017
+    """
+
+    (year, month, day, hour, minute, second) = \
+        _parseDateString(dateString, isInterval=False)
+
+    return datetime.datetime(year=year, month=month, day=day, hour=hour,
+                             minute=minute, second=second)  # }}}
+
+
+def stringToRelativeDelta(dateString, calendar='gregorian'):  # {{{
+    """
+    Given a date string and a calendar, returns an instance of
+    `MpasRelativeDelta`
+
+    Parameters
+    ----------
+    dateString : string
+        A date and time in one of the following formats:
+        - YYYY-MM-DD hh:mm:ss
+        - YYYY-MM-DD hh.mm.ss
+        - YYYY-MM-DD SSSSS
+        - DDD hh:mm:ss
+        - DDD hh.mm.ss
+        - DDD SSSSS
+        - hh.mm.ss
+        - hh:mm:ss
+        - YYYY-MM-DD
+        - YYYY-MM
+        - SSSSS
+
+        Note: either underscores or spaces can be used to separate the date
+        from the time portion of the string.
+
+    calendar: {'gregorian', 'gregorian_noleap'}, optional
+        The name of one of the calendars supported by MPAS cores
+
+    Returns
+    -------
+    relativedelta : An `MpasRelativeDelta` object
+
+    Raises
+    ------
+    ValueError
+        If an invalid `dateString` is supplied.
+
+    Author
+    ------
+    Xylar Asay-Davis
+
+    Last modified
+    -------------
+    02/04/2017
+    """
```

Comment:
An aside-- we should set something up that produces a "readthedocs" descrition of at least our public API.  Feel free to do the honors for an issue or I can also do so.  I'm happy to set this up when I get a a moment too if you would like so that we can leverage all your really good work getting docstrings into a more general sphinx-compatable format.